s3tests/functional/test_s3.py

```diff
@@ -2761,6 +2762,69 @@ def test_put_object_ifnonmatch_overwrite_existed_failed():
     eq(got_old_data, 'bar')
 
 
+@attr(resource='object')
+@attr(method='put')
+@attr(operation='append object')
+@attr(assertion='success')
+@attr('fails_on_aws')
+@attr('appendobject')
+def test_append_object():
+    bucket = get_new_bucket()
+    key = bucket.new_key('foo')
+    expires_in = 100000
+    url = key.generate_url(expires_in, method='PUT')
+    o = urlparse(url)
+    path = o.path + '?' + o.query
+    path1 = path + '&append&position=0'
+    res = _make_raw_request(host=s3.main.host, port=s3.main.port, method='PUT', path=path1, body='abc')
+    path2 = path + '&append&position=3'
+    res = _make_raw_request(host=s3.main.host, port=s3.main.port, method='PUT', path=path2, body='abc')
```

Comment:
i'm seeing these requests fail when run over https. can you please add `secure=s3.main.is_secure` as an extra argument to these calls?

Reply:
ok，I‘ve added it